contracts/Exchange.sol

```diff
@@ -155,10 +169,15 @@ contract Exchange {
         Order storage buy = buyTokenOrders[buyTokenId];
         Order storage sell = sellTokenOrders[sellTokenId];
 
-        require(buy.price >= sell.price, "buy price must be >= sell price");
+        require(buy.price >= sell.price || buy.price == 0, "buy price must be >= sell price or sell or buy price == 0");
+
+        uint publishedRate;
+        (publishedRate, ) = rates.rates(augmintToken.peggedSymbol());
+        uint buyPrice = buy.price > 0 ? buy.price : publishedRate;
+        uint sellPrice = sell.price > 0 ? sell.price : publishedRate;
 
-        // meet in the middle
-        uint price = uint(buy.price).add(sell.price).div(2);
+        // pick maker's price (whoever placed order sooner considered as maker)
```

Comment:
What prompted this change? (One could argue for either approach)

Reply:
We discussed this a while back around [this trello card](https://trello.com/c/i2g31QCE/131-exchange-match-to-use-price-of-the-earlier-order-on-match), maybe on Discord. @krosza 's view was to go with this as this the standard. I did some research too and indeed seems to be de-facto that's the standard.